diff --git a/continuousintegration/ci_runner/prepareImage.sh b/continuousintegration/ci_runner/prepareImage.sh
index a35c3a807a..ffe4681e16 100755
--- a/continuousintegration/ci_runner/prepareImage.sh
+++ b/continuousintegration/ci_runner/prepareImage.sh
@@ -37,4 +37,4 @@ mkdir -p $rpm_dir
 cp -r $rpm_source $rpm_dir
 
 echo "sudo docker build . -f continuousintegration/docker/ctafrontend/alma9/Dockerfile -t ctageneric:${image_tag}"
-sudo docker build . -f continuousintegration/docker/ctafrontend/alma9/Dockerfile -t ctageneric:${image_tag}
+podman build . -f continuousintegration/docker/ctafrontend/alma9/Dockerfile -t ctageneric:${image_tag} --network host
diff --git a/continuousintegration/orchestration/create_instance.sh b/continuousintegration/orchestration/create_instance.sh
index 964c1f6c74..4bd5aedcae 100755
--- a/continuousintegration/orchestration/create_instance.sh
+++ b/continuousintegration/orchestration/create_instance.sh
@@ -172,7 +172,7 @@ echo "Creating instance using docker image with tag: ${imagetag}"
 cp pod-* ${poddir}
 if [ ! -z "${dockerimage}" ]; then
   echo "set image to ctageneric:${imagetag}"
-  sed -i ${poddir}/pod-* -e "s/\(^\s\+image\):.*/\1: ctageneric:${imagetag}\n\1PullPolicy: Never/"
+  sed -i ${poddir}/pod-* -e "s/\(^\s\+image\):.*/\1: localhost\/ctageneric:${imagetag}\n\1PullPolicy: Never/"
 else
   sed -i ${poddir}/pod-* -e "s/\(^\s\+image:[^:]\+:\).*/\1${imagetag}/"
 fi
